### DISCUSSION TITLE
Can't update Statmic from CLI or Control Panel

### DISCUSSION THREAD

**Discussion by andyra** [statamic/cms#4793] [Troubleshooting] [answered]
I can't seem to get Statamic onto the latest version, either through the CLI or via the Control Panel UI.

**CLI**
When I run `composer update statamic/cms —with-all-dependencies ` from my project directory, everything _appears_ to install correctly; this is the output:

```
> Statamic\Console\Composer\Scripts::preUpdateCmd
Loading composer repositories with package information
Package "—with-all-dependencies" listed for update is not locked.
Updating dependencies
Nothing to modify in lock file
Installing dependencies from lock file (including require-dev)
Nothing to install, update or remove
Package fzaninotto/faker is abandoned, you should avoid using it. No replacement was suggested.
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi
Discovered Package: barryvdh/laravel-debugbar
Discovered Package: facade/ignition
Discovered Package: fideloper/proxy
Discovered Package: fruitcake/laravel-cors
Discovered Package: intervention/image
Discovered Package: laravel/tinker
Discovered Package: nesbot/carbon
Discovered Package: nunomaduro/collision
Discovered Package: rebing/graphql-laravel
Discovered Package: statamic/cms
Discovered Package: wilderborn/partyline
Package manifest generated successfully.
> @php artisan statamic:install --ansi
Addon manifest generated successfully.
Copied Directory [/vendor/statamic/cms/resources/users] To [/resources/users]
Publishing complete.
Copied Directory [/vendor/statamic/cms/resources/dist] To [/public/vendor/statamic/cp]
Publishing complete.
Compiled views cleared!
Application cache cleared!
98 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
```

**Control Panel**
I tried updating from the Updates page of the Control Panel as well, but it never seems to finish—there's a spinner in the modal that never goes away. This is the only feedback:

```
dyld: Library not loaded: /usr/local/opt/libffi/lib/libffi.7.dylib
  Referenced from: /usr/local/Cellar/php/8.0.9/bin/php
  Reason: image not found
```

I'm a little confused why `php/8.0.9` is referenced in the message above, as running `php -v` in my project directory returns `PHP 7.4.26 (cli)`. Am I missing something?

**Comment by jasonvarga** [comment]
It's possible to have a different version of PHP in the browser vs the command line.

Head to Utilities > PHP Info in the CP to check what version of PHP you're running.

**Comment by andyra** [comment]
Oh huh! The CP says I'm using `8.0.9`. I used homebrew to link to 7.4; how can I get the browser to use that as well?

**Comment by jackmcdade** [comment, marked as answer]
This always works for me when this happens https://pretagteam.com/question/phpinfo-valet-two-different-php-versions

**Reply by diskbuster** [reply]
the link does not show any information.

**Comment by andyra** [comment]
Oh brilliant, @jackmcdade, that worked for me as well. If anyone else is having this issue, here are the steps I used to install a different version of PHP on a Mac and get Valet to use it:

```
brew install php@7.4

brew unlink php@8.0
brew link php@7.4

rm~/.config/valet / valet.sock
valet restart
```

**Comment by diskbuster** [comment]
I have the same problem on a webpack.
domain uses 7.4
ssh user has 7.2 by default but be changes with ln -sfv /usr/bin/php74 /usr/bin/php

when i call the update from the CP i get told that iam using an old 7.2 and not the needed 7.4

how can i solve this?
